packages/pipeline/src/scripts/pull_slippage.ts

```diff
@@ -0,0 +1,49 @@
+import { Connection, ConnectionOptions, createConnection } from 'typeorm';
+
+import { logUtils } from '@0x/utils';
+
+import { EdpsSource} from '../data_sources/slippage';
+import { Slippage } from '../entities';
+import * as ormConfig from '../ormconfig';
+import { calculateSlippage } from '../parsers/slippage';
+import { handleError } from '../utils';
+
+// Number of orders to save at once.
+const BATCH_SAVE_SIZE = 1000;
+
+// USD amounts for slippage depths
+// tslint:disable-next-line:custom-no-magic-numbers
+const USD_AMOUNTS = [10, 100, 1000, 10000];
+
+// TODO: fetch from database
+const TOKENS = ['BAT', 'DAI', 'FUN', 'MANA', 'OMG', 'REP', 'TUSD', 'ZRX', 'MKR', 'BNB', 'USDC'];
+
+let connection: Connection;
+
+(async () => {
+    connection = await createConnection(ormConfig as ConnectionOptions);
+    const edpsSource = new EdpsSource();
+
+    logUtils.log('Fetching slippage records');
+    const nestedSlippages: Slippage[][][] = await Promise.all(TOKENS.map(async symbol => {
+        const usdPrice = await edpsSource.getUsdPriceAsync(symbol);
```

Comment:
Should `usdPrice` be the price from the cryptocompare API, or the price of some small amount from the endpoint?

Reply:
Yeah, good question. I went for CryptoCompare mostly as a pragmatic solution, because there are a few issues with using some small amount from the EDPS endpoint:

The main one is that the EDPS endpoint returns ETH prices, so unless we want to denote slippage amounts in ETH, we would still need to look up the ETH price (e.g. via CryptoCompare).

Also, prices from the EDPS are per exchange, so we would have to aggregate them to get 1 price. Definitely doable, but not sure it's worth the cost. Should we do average? Weighted average? Median? Etc.

Finally, a minor point: if we look up price via EDPS, that "small amount" (e.g. a fixed 0.01) will have *some* amount of slippage already built into it (e.g. 0.01 MKR is $5.1 while 0.01 MANA is ~$0.00035). This is not a big problem, but worth mentioning.

Reply:
Yea all good points. This might even be better. Slippage from global market price seems more useful / more important than slippage from local orderbook market price.